### ISSUE TITLE
Support private plugins

### ISSUE THREAD

**Issue by r65535** [traefik/traefik#11271]
### Welcome!

- [x] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [x] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you expect to see?

I'd like to be able to import custom plugins from private git repositories - using either a go proxy, or git/HTTPS authentication. I've tried setting envs in a traefik container for GOPROXY but that didn't work.

I don't think this is possible today? The only workaround I can find is extending the traefik docker image and downloading the plugin on top of that base image.

**Comment by Nelwhix** [comment]
Yeah recently had to debug a faulty plugin and the only solution right now is to extend the base image and download the plugin there

**Comment by r65535** [comment]
That's how I'm currently running traefik with my custom plugin.

Is there any scope in the future to allow private plugins in traefik's config.yaml (maybe through a GOPROXY or using SSH/HTTPS auth with git when pulling the go module)? I really don't want to look after a custom traefik image just to add my plugin 😄

**Comment by nmengin** [comment]
Hey @r65535.

Thanks for your suggestion.

We are interested in this issue, but we’re unsure about the use case and the traction it will receive. We are going to leave the status as kind/proposal to give the community time to let us know if they would like this idea.

We will reevaluate as people respond.

Conversation is time-boxed to 6 months.

**Comment by nmengin** [comment]
Hello,

This proposal was opened more than 6 months ago with almost no traction.
We won't add it to the Roadmap yet and we prefer to close it.

We'll re-open it later if necessary.

Thank you @r65535 for the suggestion.
